community/consistency-check/src/main/java/org/neo4j/consistency/CheckConsistencyCommand.java

```diff
@@ -142,6 +154,26 @@ public void execute( String[] args ) throws IncorrectUsage, CommandFailed
         }
     }
 
+    private void checkDbState( File storeDir, Config additionalConfiguration ) throws CommandFailed
+    {
+        try ( PageCache pageCache = StandalonePageCacheFactory.createPageCache( this.fileSystemAbstraction,
+                additionalConfiguration ) )
+        {
+            if ( new RecoveryRequiredChecker( this.fileSystemAbstraction, pageCache ).isRecoveryRequiredAt( storeDir ) )
+            {
+                throw new CommandFailed( Strings.joinAsLines(
+                        "Active logical log detected, this might be a source of inconsistencies.",
+                        "Please recover database before running the consistency check.",
+                        "To perform recovery please start database and perform clean shutdown." ) );
+            }
+        }
+        catch ( IOException e )
+        {
+            outsideWorld.stdErrLine( "Failure when checking for recovery state: '%s', continuing as normal.%n" + e
+                    .getMessage() );
```

Comment:
We'll need to print that stack trace when the debug flag is set. Perhaps OutsideWorld should sprout that capability.